File: space-game/Entities.Tests/BaseGameEntityTests.vb
Imports NSubstitute
Imports NUnit.Framework

<TestFixture>
Public Class BaseGameEntityTests

    <Test()>
    Public Sub ctor_WhenArgumentIsNull_ThrowsArgumentNullException()
        Dim entityManager As IEntityManager = Nothing

        Dim ex = Assert.Catch(Of Exception)(Function() Substitute.For(Of BaseGameEntity)(entityManager))

        Dim expectedType As Type = GetType(ArgumentNullException)

        Assert.AreEqual(ex.InnerException.GetType(), expectedType)

    End Sub

End Class

File: space-game/Entities.Tests/BaseGameEntityTests.cs
using NSubstitute;
using NUnit.Framework;
using System;

namespace Entities.Tests
{
    [TestFixture]
    public class BaseGameEntityTests
    {
        [Test]
        public void ctor_WhenArgumentIsNull_ThrowsArgumentNullException()
        {
            IEntityManager entityManager = null;
            Type expectedType = typeof(ArgumentNullException);

            Exception ex = Assert.Catch<Exception>(() => Substitute.For<BaseGameEntity>(entityManager));

            Assert.AreEqual(ex.InnerException.GetType(), expectedType);
        }
    }
}
